File: OC/JiaKaoBaoDian/JiaKaoBaoDian/JiaKaoBaoDian/ViewController.m
//
//  ViewController.m
//  JiaKaoBaoDian
//
//  Created by ligulfzhou on 2/1/16.
//  Copyright © 2016 ligulfzhou. All rights reserved.
//

#import "ViewController.h"
#import "QuestionViewController.h"

@interface ViewController ()<UITableViewDelegate, UITableViewDataSource>
{
    NSArray *staticTableData;
    NSArray *staticTableViewControllers;
    UITableView *staticTable;
}

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    staticTableData = [[NSArray alloc] initWithObjects:@"开始2016科目一", @"其他考试", @"其他考试2", nil];
    staticTableViewControllers = [[NSArray alloc] initWithObjects:[QuestionViewController class], @"其他考试", @"其他考试2", nil];
    [self setUpView];
}

-(void)setUpView
{
    staticTable = [[UITableView alloc] initWithFrame:CGRectMake(self.view.frame.origin.x, self.view.frame.origin.y, self.view.frame.size.width, 50 * [staticTableData count]) style:UITableViewStyleGrouped];
    staticTable.dataSource = self;
    staticTable.delegate = self;
    [self.view addSubview:staticTable];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


#pragma mark - tableview datasource

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"cell"];
    }
    
    cell.textLabel.text = staticTableData[indexPath.row];
    return cell;
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return [staticTableData count];
}

-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    NSLog(@"%ld - %@", indexPath.row, staticTableData[indexPath.row]);
    
    UIViewController *vc = [staticTableViewControllers[indexPath.row] new];
    [self.navigationController pushViewController:vc animated:YES];
}

@end

File: Swift/UIImageSlidesView/UIImageSlidesView/ViewController.swift
//
//  ViewController.swift
//  UIImageSlidesView
//
//  Created by ligulfzhou on 2/18/16.
//  Copyright © 2016 ligulfzhou. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {

    var tableview:UITableView!
    override func viewDidLoad() {
        super.viewDidLoad()

        self.title = "title"
        tableview = UITableView(frame: view.bounds, style: .Plain)
        tableview.registerClass(UITableViewCell.classForCoder(), forCellReuseIdentifier: "identifier")
        tableview.center = view.center
        tableview.dataSource = self
        tableview.delegate = self
        
        view.addSubview(tableview)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


    //MARK: uitableviewdatasource
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 1
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell:UITableViewCell! = tableView.dequeueReusableCellWithIdentifier("identifier")
        cell?.textLabel?.text = "查看图片"
        return cell
    }
    
    //MARK: uitableviewdelegate
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        let photoViewController = PhotosViewController()
        photoViewController.imageNames = ["B1", "B2", "B3"]
        self.navigationController?.pushViewController(photoViewController, animated: true)
    }
}

